exporter/prometheusremotewriteexporter/enhanced_logging.go

```diff
@@ -0,0 +1,112 @@
+package prometheusremotewriteexporter
+
+import (
+	"go.uber.org/zap"
+)
+
+// ErrorType represents different categories of export failures
+type ErrorType string
```

Comment:
please do not export structs

Reply:
Using strings as error codes is not ideal considering that these could be "enums" instead, or have a better wrapping around them. 

What I suggest if you do want to follow this path is having something like:
```golang 
type exportError struct {
   classification classification
   err error
}

func (ee exportError) Unwrap() error { return ee.err }
```

This will allow you to wrap the error you got, allow it to be used with `errors.Is`, and allow you to expose the classification.